src/Middleware/Diagnostics/src/ExceptionHandler/ExceptionHandlerExtensions.cs

```diff
@@ -95,7 +99,22 @@ public static IApplicationBuilder UseExceptionHandler(this IApplicationBuilder a
                 throw new ArgumentNullException(nameof(options));
             }
 
-            return app.UseMiddleware<ExceptionHandlerMiddleware>(Options.Create(options));
+            return app.Use(next =>
+            {
+                var loggerFactory = app.ApplicationServices.GetRequiredService<ILoggerFactory>();
+                var diagnosticListener = app.ApplicationServices.GetRequiredService<DiagnosticListener>();
+                var endpointDataSource = app.ApplicationServices.GetRequiredService<EndpointDataSource>();
+
+                if (!string.IsNullOrEmpty(options.ExceptionHandlingPath) && options.ExceptionHandler is null)
+                {
+                    var errorBuilder = app.New();
+                    errorBuilder.UseRouting(overrideEndpointRouteBuilder: false);
+                    errorBuilder.Run(next);
+                    options.ExceptionHandler = errorBuilder.Build();
```

Comment:
This is creating a new instance of options, applicationBuilder, request delegate and the middlewares per request. Why not moving this code in `ExceptionHandlerMiddleware` ctor? The downside is that `IApplicationBuilder` must be added to its ctor parameters which is a breaking change.